### PR TITLE
Ea 3246 lang filter on title improved

### PR DESCRIPTION
**Pull request by gsergiu**
recreated PR to clearly see the changes

### CODE REVIEW

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java [hunk 1]
```diff
@@ -70,10 +70,10 @@ private UserSetQuery buildSearchQuery(Map<String, String> searchCriteria, String
     private void addTitleLangCriterion(Map<String, String> searchCriteria, UserSetQuery searchQuery)
         throws ParamValidationException {
     if (searchCriteria.containsKey(WebUserSetFields.LANG)) {
-        String lang = searchCriteria.get(WebUserSetFields.LANG);
-        if (!"en".equals(lang)) {
+        String lang = searchCriteria.get(WebUserSetFields.LANG).toLowerCase();
+        if (! UserSetUtils.is2Letter(lang)) {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java:74]: please replace the implementation for validating the value of the language against the real ISO Language codes. 
Available in the Locale.getISOLanguages()

File: set-definitions/src/main/java/eu/europeana/set/definitions/model/utils/UserSetUtils.java [hunk 2]
```diff
@@ -150,6 +150,21 @@ public static boolean isInteger(String setId) {
...
         }
     }
+ 
+    /**
+     * This method is to check if a string is a 2-letter string
+     *
+     * @param string
+     * @return true if 2-letter string
+     */
+    public static boolean is2Letter(String string) {
```
Comment (gsergiu) [inline set-definitions/src/main/java/eu/europeana/set/definitions/model/utils/UserSetUtils.java:160]: this method is not needed

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java (hunk 1 above)
Comment (gsergiu) [reply set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java:74]: please use the same solution from the entity management https://github.com/europeana/entity-management/blob/cb34d4dd448a4328d0cc57d2ee3dfb291f352a24/entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsCleaner.java#L41

Comment (gsergiu) [reply set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java:74]: A new class LanguageUtils can be implemented in the api coommons in this package: eu.europeana.api.commons.definitions.utils

**Review by gsergiu — COMMENTED** [review]
the language validation needs to be improved

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java (hunk 1 above)
Comment (SrdjanStevanetic) [reply set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java]: Improved.

**Pull request merged**